### ISSUE TITLE
Widgets gone after update to 7.4

### ISSUE THREAD

**Issue by abonhote** [TBog/TBLauncher#401]
# Description #
After updating to 7.4, my two home widgets disappeared. They seem to be registered, I can chose to remove/move/resize them, but apart from remove, nothing seems to do anything. No border appears.

I can remove the widgets, but re-adding them just adds them to the registry. They're not visible.

## Context ##
Upgrade 7.4, widgets.

https://youtube.com/shorts/nhrkigodtFk?si=tirZoy_23O6y5mhR

## Steps to Reproduce ##
- install pre 7.4
- add some widgets
- update to 7.4
- find the widgets

## Device info ##

  * Device: Samsung Galaxy S23 Ultra
  * OS: Android 13
  * TinyBit Launcher version: 7 4

**Comment by bencew** [comment]
I've figured out that opening an app from the app list in "Search", then hitting the Home button brings up my widgets. The Home button is set to "Show <<Widget>> desktop".
Then if I switch between "Search" and "Show <<Widget>> desktop" by tapping on the screen, the widgets disappear again.

**Comment by abonhote** [comment]
Not an android dev, but could it be an issue related to screen resolution? Whenever I change the resolution, my widgets get lost/moved around, even with 7.3. Maybe one of the bumps contained changes in that area.

Just guessing...

**Comment by abonhote** [comment]
As this ticket is not marked as **bug**, is there something I am doing wrong?

**Comment by JukeboxZulu** [comment]
I am also having this issue, widgets can still be clicked on but are invisible. Using my home button (set to Toggle Search/Widget) does not show the widgets. While in Search, using back button (set to Show Widgets) does show widgets.

Hope this helps to debug.

**Comment by TBog** [comment]
Interesting. I'll try to revert any changes related to the widget screen.
I'll also add more debug information and reset options for widgets so we don't accidentally place them outside of the screen.

**Comment by TBog** [comment]
From the video I see that `<widget><size width="750" height="330" /><position
left="0" top="0" screen="0" />< widget>`
You may have to scroll left to see the widget.

**Comment by abonhote** [comment]
I have disabled pages, I can't scroll.

![Screenshot_20230920_072329_TinyBit launcher.jpg](https://github.com/TBog/TBLauncher/assets/264473/04d15a51-4af6-4101-815d-9a82747e669b)

In 7.3, this is my topmost widget, working perfectly:

![Screenshot_20230920_072507_TinyBit launcher.png](https://github.com/TBog/TBLauncher/assets/264473/aef6e165-d372-4054-9ecb-f311e1d265fc)

**Comment by TBog** [comment]
> I have disabled pages, I can't scroll.

Enable pages as a workaround

**Comment by abonhote** [comment]
Not really. I enabled pages and upgraded to 7.4 from 7.3. Hope this helps:

https://youtube.com/shorts/anh4TR4Cod4?si=6fzRC82modKWfVb4

The widgets are still those from my 7.3 setup.

**Comment by TBog** [comment]
![Screenshot_20230920_112721.jpg](https://github.com/TBog/TBLauncher/assets/6465804/af7b8c18-4fb8-44bf-beab-b90609bff7a9)

this should be the desktop that the launcher starts with

**Comment by abonhote** [comment]
```xml
<preference key="initial-desktop" value="showWidgets" />
```

Should be ok, right? I could attach my whole config xml if you like.

**Comment by opk12** [comment]
Same issue. I have Android 7.1 + TB 7.4 from F-droid. I have enabled right-left pages (but not top-bottom) and all options from `Wallpaper interactions`, except `Send touch events` and `Emulate drag event`.

Do pages still work? I have widget A in the left page and widget B in the center page, I click Home button (set to show wallpaper), I scroll to the left page, I tap on the black screen, I expect app A to open, instead app B opens.

After updating, I added a widget as a test. Now, when I open a Fennec PWA or OsmAnd from the Search results or the dock and they take a longer time to open, TB shows the center page with its widgets during that fraction of a second - too quickly for me to try to interact. But my initial desktop is `Desktop mode <<Search>>`.

**Comment by Saijin-Naib** [comment]
I'm a new user starting from 7.4 and I also experience the widgets being transparent but still able to be clicked or otherwise interacted with.

A way to reproduce the behavior is to set home button to Widget Desktop Center, and back to Search Desktop, and use a flick gesture to Toggle Between Search and Widget.

Pressing home results in ghost widgets, clearing search menu and flicking with the toggle makes them appear.

**Comment by opk12** [comment]
I have no problem with widgets since I updated to 7.5 a long time ago.

**Comment by abonhote** [comment]
Yup, you're right. I'll close this one.
